src/openrct2-ui/windows/ServerStart.cpp

```diff
@@ -47,291 +47,274 @@ enum {
-            window_start_textbox(w, widgetIndex, STR_STRING, _name, 64);
-            break;
-        case WIDX_DESCRIPTION_INPUT:
-            window_start_textbox(w, widgetIndex, STR_STRING, _description, MAX_SERVER_DESCRIPTION_LENGTH);
-            break;
-        case WIDX_GREETING_INPUT:
-            window_start_textbox(w, widgetIndex, STR_STRING, _greeting, CHAT_INPUT_SIZE);
-            break;
-        case WIDX_PASSWORD_INPUT:
-            window_start_textbox(w, widgetIndex, STR_STRING, _password, 32);
-            break;
-        case WIDX_MAXPLAYERS_INCREASE:
-            if (gConfigNetwork.maxplayers < 255)
-            {
-                gConfigNetwork.maxplayers++;
-            }
-            config_save_default();
-            w->Invalidate();
-            break;
-        case WIDX_MAXPLAYERS_DECREASE:
-            if (gConfigNetwork.maxplayers > 1)
-            {
-                gConfigNetwork.maxplayers--;
-            }
-            config_save_default();
-            w->Invalidate();
-            break;
-        case WIDX_ADVERTISE_CHECKBOX:
-            gConfigNetwork.advertise = !gConfigNetwork.advertise;
-            config_save_default();
-            w->Invalidate();
-            break;
-        case WIDX_START_SERVER:
-            network_set_password(_password);
-            WindowScenarioselectOpen(WindowServerStartScenarioselectCallback, false);
-            break;
-        case WIDX_LOAD_SERVER:
-            network_set_password(_password);
-            auto intent = Intent(WC_LOADSAVE);
-            intent.putExtra(INTENT_EXTRA_LOADSAVE_TYPE, LOADSAVETYPE_LOAD | LOADSAVETYPE_GAME);
-            intent.putExtra(INTENT_EXTRA_CALLBACK, reinterpret_cast<void*>(WindowServerStartLoadsaveCallback));
-            context_open_intent(&intent);
-            break;
+        switch (widgetIndex)
+        {
+            case WIDX_CLOSE:
+                window_close(this);
+                break;
+            case WIDX_PORT_INPUT:
+                window_start_textbox(this, widgetIndex, STR_STRING, _port, 6);
+                break;
+            case WIDX_NAME_INPUT:
+                window_start_textbox(this, widgetIndex, STR_STRING, _name, 64);
+                break;
+            case WIDX_DESCRIPTION_INPUT:
+                window_start_textbox(this, widgetIndex, STR_STRING, _description, MAX_SERVER_DESCRIPTION_LENGTH);
+                break;
+            case WIDX_GREETING_INPUT:
+                window_start_textbox(this, widgetIndex, STR_STRING, _greeting, CHAT_INPUT_SIZE);
+                break;
+            case WIDX_PASSWORD_INPUT:
+                window_start_textbox(this, widgetIndex, STR_STRING, _password, 32);
+                break;
+            case WIDX_MAXPLAYERS_INCREASE:
+                if (gConfigNetwork.maxplayers < 255)
+                {
+                    gConfigNetwork.maxplayers++;
+                }
+                config_save_default();
+                OnPrepareDraw();
```

Comment:
This is unfortunately not the same. `Invalidate()` is the correct function to call